### PR TITLE
models( API/models) Create the models for DB

### PR DESCRIPTION
**Pull request by BenFergusonSA**
Created the required models for the mongoDB

### CODE REVIEW

File: models/organizer.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+var {organizerEventSchema} = require('./organizerEvent');
+
+var organizerSchema = mongoose.Schema({
+    name: {
+        type: String,
+        required: true,
+        trim: true,
+        minlength: 1
+    },
+    organizerEvents: [ organizerEventSchema ]
```
Comment (Albertcoetzer) [inline models/organizer.js]: rather do this:
organizerEvents: {
    type: [{
        ref: 'OrganizerEvent',
        required: true,
        type: mongoose.Schema.Types.ObjectId
     }]
}

File: models/organizerEvent.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        type: String,
+        required: true,
+        trim: true,
+        minlength: 1
+    },
+    date: {
+        type: Date,
+        required: true,
+    },
+    staff: [ staffSchema ],
```
Comment (Albertcoetzer) [inline models/organizerEvent.js]: Same over here

File: models/organizer.js (hunk 1 above)
Comment (BenFergusonSA) [reply models/organizer.js]: Should I change the actual organizerEvent Schemas ?

File: models/ticket.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    used: {
+        type: Boolean,
+        required: true,
+        default: false
+    },
+    purchasedOn: {
+        type: Date,
+        required: true
+    },
+    ticketType: ticketSchema
```
Comment (Albertcoetzer) [inline models/ticket.js]: Referencing the wrong model.
And here just without the [] so
ticketType: {
ref: 'TicketType',
required: true
}

File: models/staff.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+var mongoose = require('mongoose');
+var {userSchema} = require('./user');
+
+var staffSchema = userSchema;
```
Comment (Albertcoetzer) [inline models/staff.js]: Make it their own schema so we can add permissions variables later on

File: models/organizerEvent.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    TicketTypes: [ ticketTypeSchema ]
```
Comment (Albertcoetzer) [inline models/organizerEvent.js]: And here

File: models/organizer.js (hunk 1 above)
Comment (BenFergusonSA) [reply models/organizer.js]: My mistake, your code makes the reference to an existing organizer rather than creating copies.

Comment (Albertcoetzer) [reply models/organizer.js]: Exactly. We save the organiser in the db and just reference it here via ObjectId

**Comment by BenFergusonSA** [comment]
Reviewed code fixed.

**Pull request merged**